main.go

```diff
@@ -32,8 +33,77 @@ import (
 	"github.com/spf13/cast"
 
 	"github.com/bank-vaults/secret-init/provider"
+	"github.com/bank-vaults/secret-init/provider/file"
 )
 
+func NewProvider(providerName string) (provider.Provider, error) {
+	switch providerName {
+	case file.ProviderName:
+		provider, err := file.NewProvider(os.DirFS("/secrets"))
```

Comment:
You can use root path since we want to be able to specify paths relative to that
```suggestion
		provider, err := file.NewProvider(os.DirFS("/"))
```